db/version_set.cc

```diff
@@ -356,50 +355,50 @@ Status Version::Get(const ReadOptions& options, const LookupKey& k,
       state->last_file_read = f;
       state->last_file_read_level = level;
 
-      Saver saver;
-      saver.state = kNotFound;
-      saver.ucmp = state->ucmp;
-      saver.user_key = state->user_key;
-      saver.value = state->value;
-
-      Status s = state->vset->table_cache_->Get(*state->options, f->number,
+      state->s = state->vset->table_cache_->Get(*state->options, f->number,
                                                 f->file_size, state->ikey,
-                                                &saver, SaveValue);
-      if (!s.ok()) {
-        state->s = s;
+                                                &state->saver, SaveValue);
+      if (!state->s.ok()) {
+        state->found = true;
         return false;
       }
-      switch (saver.state) {
+      switch (state->saver.state) {
         case kNotFound:
           return true;  // Keep saerching in other files
         case kFound:
-          state->s = s;
+          state->found = true;
           return false;
         case kDeleted:
           return false;
         case kCorrupt:
           state->s = Status::Corruption("corrupted key for ", state->user_key);
+          state->found = true;
           return false;
       }
     }
   };
 
   State state;
-  state.s = Status::NotFound(Slice());
+  state.found = false;
   state.stats = stats;
   state.last_file_read = nullptr;
   state.last_file_read_level = -1;
 
   state.options = &options;
-  state.ikey = ikey;
-  state.user_key = user_key;
-  state.ucmp = ucmp;
+  state.ikey = k.internal_key();
+  state.user_key = k.user_key();
+  state.ucmp = vset_->icmp_.user_comparator();
```

Comment:
It looks like the following data is being placed both in state and state.saver.  I don't think that is necessary.  Can you drop the state fields that are duplicates of fields in Saver?

ucmp
user_key

Reply:
Right on it, maybe also drop the value field in State?